src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs

```diff
@@ -378,9 +378,19 @@ private bool HandleNamedTypesWorker(INamedTypeSymbol x, INamedTypeSymbol y, Dict
                         return false;
                     }
 
+                    var tupleNamesMustMatch = _symbolEquivalenceComparer._tupleNamesMustMatch;
+
                     for (var i = 0; i < xElements.Length; i++)
                     {
-                        if (!AreEquivalent(xElements[i].Type, yElements[i].Type, equivalentTypesWithDifferingAssemblies))
+                        var xElement = xElements[i];
+                        var yElement = yElements[i];
+                        if (tupleNamesMustMatch &&
+                            xElement.Name != yElement.Name)
```

Comment:
Do we care about VB and case-insensitive identifier names? (I can see this going both ways; I don't have a preconceived answer here.)

Reply:
Good question.  I followed the rest of this function.  So i'll keep consistent for now.

Reply:
the equivalence comparer doesn't seem to handle this today.  so i'm not going to change this.